File: backend/app/services/case_service.py
from typing import List
from app.models.case import Case, CaseStatistics


class CaseService:
    """Service for managing legal cases"""
    
    def get_all_cases(self) -> List[Case]:
        """Retrieve all cases"""
        # Implementation will be added in later tasks
        pass
    
    def get_case_by_id(self, case_id: str) -> Case:
        """Retrieve a specific case by ID"""
        # Implementation will be added in later tasks
        pass
    
    def get_cases_by_type(self, case_type: str) -> List[Case]:
        """Retrieve cases filtered by type"""
        # Implementation will be added in later tasks
        pass
    
    def get_case_statistics(self) -> CaseStatistics:
        """Get dashboard statistics"""
        # Implementation will be added in later tasks
        pass

File: frontend/src/services/caseService.ts
import { apiClient } from './api';
import { Case, CaseStatistics } from '../types/api';

export const caseService = {
  async getAllCases(): Promise<Case[]> {
    const response = await apiClient.get<Case[]>('/api/cases');
    return response.data;
  },

  async getCaseById(caseId: string): Promise<Case> {
    const response = await apiClient.get<Case>(`/api/cases/${caseId}`);
    return response.data;
  },

  async getCaseStatistics(): Promise<CaseStatistics> {
    const response = await apiClient.get<CaseStatistics>('/api/cases/statistics');
    return response.data;
  },
};